drivers/storage/portworx/component/portworx_basic.go

```diff
@@ -616,7 +616,7 @@ func (c *portworxBasic) refreshTokenSecret(secret *v1.Secret, cluster *corev1.St
 func generatePxSaToken(cluster *corev1.StorageCluster, expirationSeconds int64) (*authv1.TokenRequest, error) {
 	tokenRequest := &authv1.TokenRequest{
 		Spec: authv1.TokenRequestSpec{
-			Audiences:         []string{"px"},
+			Audiences:         []string{"https://kubernetes.default.svc.cluster.local"}, // the k8s api server service
```

Comment:
we should probably figure this out dynamically as some customers customize this URL

Reply:
You can leave the Audiences unspecified. It will default to the API server.

```
dev-neelesh-thakur3:~/oc$ kubectl create token -n kube-system portworx -v8
I0709 18:49:27.174112 2761203 loader.go:395] Config loaded from file:  ./kubeconfig.neelesh
I0709 18:49:27.174834 2761203 request.go:1212] Request Body: {"kind":"TokenRequest","apiVersion":"authentication.k8s.io/v1","metadata":{"creationTimestamp":null},"spec":{"audiences":null,"expirationSeconds":null,"boundObjectRef":null},"status":{"token":"","expirationTimestamp":null}}
I0709 18:49:27.174905 2761203 round_trippers.go:463] POST https://api.pwx-ocp-152-217.pwx.purestorage.com:6443/api/v1/namespaces/kube-system/serviceaccounts/portworx/token
I0709 18:49:27.174912 2761203 round_trippers.go:469] Request Headers:
I0709 18:49:27.174921 2761203 round_trippers.go:473]     Accept: application/json, */*
I0709 18:49:27.174929 2761203 round_trippers.go:473]     Content-Type: application/json
I0709 18:49:27.174934 2761203 round_trippers.go:473]     User-Agent: kubectl/v1.28.0 (linux/amd64) kubernetes/855e7c4
I0709 18:49:27.174943 2761203 round_trippers.go:473]     Authorization: Bearer <masked>
I0709 18:49:27.200252 2761203 round_trippers.go:574] Response Status: 201 Created in 25 milliseconds
I0709 18:49:27.200279 2761203 round_trippers.go:577] Response Headers:
I0709 18:49:27.200288 2761203 round_trippers.go:580]     Content-Type: application/json
I0709 18:49:27.200296 2761203 round_trippers.go:580]     Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
I0709 18:49:27.200305 2761203 round_trippers.go:580]     X-Kubernetes-Pf-Flowschema-Uid: 36cc48bd-4dfd-4d7e-a604-4eacbc1c2371
I0709 18:49:27.200337 2761203 round_trippers.go:580]     X-Kubernetes-Pf-Prioritylevel-Uid: 3f379663-f565-4a3b-ad9f-2f6bb4807478
I0709 18:49:27.200347 2761203 round_trippers.go:580]     Content-Length: 1747
I0709 18:49:27.200355 2761203 round_trippers.go:580]     Date: Wed, 10 Jul 2024 00:49:27 GMT
I0709 18:49:27.200448 2761203 round_trippers.go:580]     Audit-Id: 8e20d9db-7e90-489b-8a1e-ab888a675383
I0709 18:49:27.200459 2761203 round_trippers.go:580]     Cache-Control: no-cache, private
I0709 18:49:27.200493 2761203 request.go:1212] Response Body: {"kind":"TokenRequest","apiVersion":"authentication.k8s.io/v1","metadata":{"name":"portworx","namespace":"kube-system","creationTimestamp":"2024-07-10T00:49:27Z","managedFields":[{"manager":"kubectl","operation":"Update","apiVersion":"authentication.k8s.io/v1","time":"2024-07-10T00:49:27Z","fieldsType":"FieldsV1","fieldsV1":{"f:spec":{"f:expirationSeconds":{}}},"subresource":"token"}]},"spec":{"audiences":["https://kubernetes.default.svc"],"expirationSeconds":3600,"boundObjectRef":null},"status":{"token":"...
```

Reply:
Leave it empty works.